### ISSUE TITLE
Dealing with resource allocation errors

### ISSUE THREAD

**Issue by myndzi** [myndzi/pool2#7]
The current behavior of pool2 is as follows:

A pool is initialized with 'live = false'

When a resource is successfully allocated, 'live' is set to 'true'

When a resource allocation fails, if 'live' is 'false', the pool emits an error and destroys itself. This is intended to deal with the circumstance where a pool is initialized with an invalid allocator -- that is, resources are expected never to be allocated correctly (db password wrong, say).

Otherwise, when 'live' is 'true', allocation errors will be ignored and new allocations will be attempted if necessary (to fill the minimum pool size, or to serve pending allocation requests)

There is a deliberate decision to separate the process of acquiring a resource to fill a request and the process of serving a request; this makes it difficult or undesirable to reject specific resource requests when specific allocations fail.

Is this behavior reasonable and desired?

Should 'error' events be emitted every time an allocation fails? This behavior would require bookkeeping code from users of pool2 to prevent crashes.

What about an alternate event, such as 'warn'?

Or, is the 'retry until success' model acceptable?

**Comment by myndzi** [comment]
As mentioned in the knex thread, there isn't currently any setting to cause a pending allocation request to time out. (The acquireTimeout option applies to acquiring resources from the source to the pool, not acquiring resources from the pool to a client)

I'm not sure whether it makes sense for this to be the pool's responsibility, but if it isn't, there definitely needs to be a way to _cancel_ a request, which there isn't.

One argument in favor of this is that it's simply confusing (I deleted my comment, thinking I was mistaken, when I realized I had tests about "timing out on acquire") to manage timeouts for two legs of the resource acquisition process.

Should an option be added to time out a client's acquire request, or should a means be provided (similar to setTimeout/clearTimeout) to cancel a pending request?
